### PR TITLE
feat(protocol): add serverless protocol library

### PR DESCRIPTION
**Pull request by mmoreiradj**
Create a lib to communicate over a serial port.

Added README on how to test it.

I think one point of improvement is to not calculate the checksum from the payload converted to json as we don't know wether serde or another json converter parser will convert it the same way.

### CODE REVIEW

**Comment by mmoreiradj** [comment]
Added libudev-dev in github actions step

**Comment by remi-espie** [comment]
The checksum is now calculated using the content of the payload and the message type, without relying on `serde_json::to_string`

File: src/serverless_protocol/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+1. Use socat to create a virtual serial port:
+
+```bash
+socat -d -d pty,raw,echo=0 pty,raw,echo=0
+```
+
+2. Run the example:
+
+```bash
+cargo run --example cargo run --example serverless_protocol_example -- --serial-path-a=<path_to_first_pty> --serial-path-b=<path_to_second_pty>
```
Comment (remi-espie) [inline src/serverless_protocol/README.md]: Remove typo
```suggestion
cargo run --example serverless_protocol_example -- --serial-path-a=<path_to_first_pty> --serial-path-b=<path_to_second_pty>
```

**Review by remi-espie — APPROVED** [review]
LGTM

**Comment by mmoreiradj** [comment]
@charley04310 @thomas-mauran does this match your needs ?

**Comment by mmoreiradj** [comment]
closed in favor of [this](https://github.com/virt-do/cloudlet/pull/20), we dont need this protocol, we can just use the network now

**Pull request closed without merging**